xpsi/PostProcessing/_runs.py

```diff
@@ -122,10 +125,57 @@ def _transform(q, **kwargs):
             _transform = None
             _overwrite = False
 
+        if multi_mode:
+            for vec in range(len(roots)):
+                filerootpath =_os.path.join(base_dirs[vec], roots[vec])
+                # Read the file and manually handle blank lines
+                with open(filerootpath+"post_separate.dat", 'r') as file:
+                    lines = file.readlines()
+
+                # Convert lines to a numpy array, replacing blank lines with a marker (e.g., None or NaN)
+                data = []
+                for line in lines:
+                    stripped = line.strip()
+                    if stripped:  # Non-blank line
+                        data.append(_np.array(stripped.split(), dtype=float))
+                    else:  # Blank line
+                        data.append(None)
+
+                # Separate modes based on None markers
+                modes = []
+                current_mode = []
+                for row in data:
+                    if row is None:
+                        if current_mode:
+                            modes.append(_np.array(current_mode))
+                            current_mode = []
+                    else:
+                        current_mode.append(row)
+
+                # Append the last mode if not empty
+                if current_mode:
+                    modes.append(_np.array(current_mode))
+
+                for mode in range(len(modes)):
+                    _np.savetxt(filerootpath+f"mode{mode+1}.txt", modes[mode])
+                    roots.append(roots[vec]+f"mode{mode+1}")
+                    run_IDs.append(run_IDs[vec]+f"{mode_label} {mode+1}")
+                    base_dirs.append(base_dirs[vec])
+                    use_nestcheck.append(use_nestcheck[vec])
+            # Forget about the defaul xpsi loaded file
+            roots = roots[1:]
+            run_IDs = run_IDs[1:]
+            base_dirs = base_dirs[1:]
+            use_nestcheck = use_nestcheck[1:]
+
+            cls.mode_len =len(modes)
```

Comment:
Could it it be this line that is the problem? Is this saving the number of modes in some model specific variable, or to something that gets overwritten every time a new model is loaded?

Reply:
I guess the `mode_len` should be run-specific, since even different runs can have different number of modes.

Reply:
So I think there are a few issues. First of all, the number of modes (and the other mode info too) is saved to a class variable that gets overwritten every time runs to new models are loaded. Secondly, this does not seem to work (at least in a general case) either if loading many runs for the same model, since it assumes that all the runs have the same number of modes.


I am not sure if there is an easy way to fix it, but maybe we can just raise errors/warnings if somebody tries to use this with multiple models/runs, or at least warn about this in the docstring.

Reply:
A solution for the 1st issue would be to make multi-modes variables (like `mode_len`) not class variables. This can be achieved by replacing `return cls(runs, likelihood, ID, **kwargs)` by something like
`Runs = cls(runs, likelihood, ID, **kwargs)`
`Runs.mode_len =len(modes)`
`return Runs`

For the second issue, this can be fixed  by storing the `mode_len` and other such values in form of a dictionary, with the keys being the run ID and the values the `mode_len` ? 

If you agree I can do it quickly

Reply:
Yes, those sounds like good ideas to. Please go ahead if you like. I think you should then also change `mode_label` to not be a class variable either (so that different models can have different labels).

And I guess we need to assume that if we set multi_mode=True, all the loaded runs should have been run using the MultiNest set to `mmodal=True`?